scripts/activate-venv.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# run with 'source scripts/activate-venv.sh'
+set -e
+
+ROOT="$(cd "$(dirname "$0")"/.. && pwd)"
+ENV_NAME=".py_venv"
+
+# if venv does not exist, send error message and exit
+if [ ! -d "${ROOT}/${ENV_NAME}" ]; then
+  echo "Virtual environment '${ENV_NAME}' not found."
+  if [[ $OSTYPE == "linux-gnu"* ]]; then
+    echo "run 'sudo ./scripts/install-build-tools.sh' to create it."
+  elif [[ $OSTYPE == "darwin"* ]]; then
+    echo "run './scripts/install-build-tools.sh' to create it."
+  fi
+  exit 1
+fi
+
+# activate virtual environment
+if source ${ENV_NAME}/bin/activate; then
```

Comment:
This script tries to detect the environment regardless of what directory it is being called from; it does so by locating the `ROOT` via the `dirname` of the script itself.

However, this `source` assumes the environment is in the current working directory. Probably, this should be changed to `source "$ROOT/$ENV_NAME/bin/activate"`.

Reply:
a few places in this PR do this; worth making sure the environment is being sourced correctly everywhere. That's probably easiest by using the same convention everywhere to access it.

Reply:
point well made, updated

Reply:
Not all instances have been caught here. Unresolving till they are handled.